src/config.rs

```diff
@@ -117,59 +117,55 @@ pub fn home_dir() -> Utf8PathBuf {
 pub struct PerfConfig {
     /// Parallelism settings
     pub parallelism: ParallelConfig,
-    /// Witness generation settings
-    pub witness_generation: WitnessGeneration,
 }
 
 impl PerfConfig {
     fn fully_sequential() -> Self {
         Self {
             parallelism: ParallelConfig {
-                recursive_steps: Flow::Sequential,
-                synthesis: Flow::Sequential,
-                poseidon_witnesses: Flow::Sequential,
-            },
-            witness_generation: WitnessGeneration {
-                precompute_neptune: false,
+                wit_gen_vs_folding: Flow::Sequential,
+                frames: Flow::Sequential,
+                slots: Flow::Sequential,
             },
         }
     }
 
-    fn max_parallel_simple() -> Self {
+    fn parallel_wit_gen_vs_folding() -> Self {
         Self {
             parallelism: ParallelConfig {
-                recursive_steps: Flow::Parallel,
-                synthesis: Flow::Parallel,
-                poseidon_witnesses: Flow::Parallel,
-            },
-            witness_generation: WitnessGeneration {
-                precompute_neptune: true,
+                wit_gen_vs_folding: Flow::Parallel,
+                frames: Flow::Sequential,
+                slots: Flow::Sequential,
             },
         }
     }
 
-    fn parallel_synthesis() -> Self {
+    fn parallel_slots() -> Self {
         Self {
             parallelism: ParallelConfig {
-                recursive_steps: Flow::Parallel,
-                synthesis: Flow::Parallel,
-                poseidon_witnesses: Flow::Sequential,
-            },
-            witness_generation: WitnessGeneration {
-                precompute_neptune: true,
+                wit_gen_vs_folding: Flow::Sequential,
+                frames: Flow::Sequential,
+                slots: Flow::Parallel,
             },
         }
     }
 
-    fn parallel_steps_only() -> Self {
+    fn parallel_frames() -> Self {
```

Comment:
This is a great initiative! Could we take this occasion to add a doc comment on each of these?